QuizApp/Data/Common/Networking/BaseNetworkClient.swift

```diff
@@ -0,0 +1,183 @@
+import Foundation
```

Comment:
I would reorganize this file a bit:

1. HttpMethod, NetworkError, NetworkResponseStatus and ClientError should be in a separate file (for example `BaseClientModels`
2. BaseNetworkClientProtocol also in a separate file.
3. I wouldn't implement `BaseNetworkClientProtocol` in an extension, but rather in the original implementation and I'd put it on top. That way when you open `BaseNetworkClient` first functions you'd see are those that are called from the outside and the rest of the logic is "hidden" lower in the file.

Also, you're using `async` and `await` in places where it's not necessary, like:
```swift
private func encodeRequestBody<T: Encodable>(model: T) throws -> Data {
    try encoder.encode(model)
}
```
where the signature of the `encode` function is:
```swift
open func encode<T>(_ value: T) throws -> Data where T : Encodable
```
which in no way indicates that it's async.

And base client logic should go to `QuizApp/Core` folder, while the exact clients should be in `QuizApp/Feature/Client` or `QuizApp/Client/Feature/` which ever you prefer.